### DISCUSSION TITLE
zfs module and prog versions missmatch after build from source on ubuntu 20.04

### DISCUSSION THREAD

**Discussion by dipol0** [openzfs/zfs#13035] [Q&A]
Hi. Sorry my english.

OS Ubuntu 20.04 lts
gcc 10.3.0

I build zfs from source use:

_git clone https://github.com/openzfs/zfs.git_

[As in documentation](https://openzfs.github.io/openzfs-docs/Developer%20Resources/Building%20ZFS.html?highlight=build) i make:

_sh ./autogen.sh
./configure_

> !!! _cat zfs.release_
> get - _2.1.99-716_g7633c0aed_

Ok. Next... I integrate ZFS to Linux kernel sources (v.5.5.17) and build new kernel. Continue in ZFS...

_make -s -j$(nproc)
sudo make install; sudo ldconfig; sudo depmod_

Install new kernel, reboot.... And zfs-volumes not mounted. I check prog and modules version:

_zpool version
zfs-2.1.99-705_g8d6a598a2
zfs-kmod-2.1.99-716_g7633c0aed_

Fist i think may be previous installation not correct or some thing...
Go to source:

_cd ~/src/zfs/cmd/zpool/.libs$
./zpool version_

and get same:

_zfs-2.1.99-705_g8d6a598a2
zfs-kmod-2.1.99-716_g7633c0aed_

**what wrong?**

i remove zfs 

_sudo make uninstall; sudo ldconfig; sudo depmod_

i remove sources and repeat all steps from git clone.... but it made made kernel module with new verzion zfs but zpool, zfs and etc create wrong (old) version. May be i mistake somewhere ?

Tnx.

UPD:

if i run warper:

_cd ~/src/zfs/cmd/zpool/$
./zpool version_

then show correct version _zfs-2.1.99-716_g7633c0aed_

**Comment by rincebrain** [comment]
I think a consensus was reached on this in #13045.
